doc/nrf/gs_modifying.rst

```diff
@@ -84,54 +93,115 @@ The following CMake commands can be managed by SES, if they target the ``app`` l
 
-The |VSC| extension lets you modify your build configuration for custom boards, add additional CMake build arguments, select Kconfig fragments, and more.
-For detailed instructions, see the `nRF Connect for Visual Studio Code`_ documentation site.
+.. _configure_application_hw:
 
-Changing the configuration permanently
-======================================
+Hardware-related configuration
+------------------------------
 
-To configure your application and maintain the configuration when you clean the build directory, add your changes to the :file:`prj.conf` file in your application directory.
-In this file, you can specify different values for configuration options that are defined by a library or board, and you can add configuration options that are specific to your application.
+.. ncs-include:: guides/build/index.rst
+   :docset: zephyr
+   :dedent: 3
+   :start-after: Devicetree
+   :end-before: The preprocessed devicetree sources
 
-See :ref:`zephyr:setting_configuration_values` in the Zephyr documentation for information on how to edit the :file:`prj.conf` file.
+The preprocessed devicetree sources (stored in :file:`*.dts.pre.tmp`) are parsed by :file:`zephyr/scripts/dts/gen_defines.py` to generate a :file:`devicetree_unfixed.h` header with preprocessor macros.
 
-If you work with SES, the :file:`prj.conf` file is read when you open a project.
-This means that after you edit this file, you must re-open your project.
+.. ncs-include:: guides/build/index.rst
+   :docset: zephyr
+   :dedent: 3
+   :start-after: header with preprocessor macros.
+   :end-before: This file is just for reference.
 
-.. note::
-   It is possible to change the default configuration for a library by changing the :file:`Kconfig` file of the library.
-   However, best practice is to override the configuration in the application configuration file :file:`prj.conf`.
+The :file:`zephyr.dts` file describes the entire hardware-related configuration of the system, while the header file provides defines that can be used to refer hardware configuration from the source code.
+
+For more detailed information, see Zephyr's :ref:`zephyr:dt-guide`.
+
+.. _configure_application_sw:
+
+Software-related configuration
+------------------------------
+
+.. ncs-include:: guides/build/index.rst
+   :docset: zephyr
+   :dedent: 3
+   :start-after: Kconfig
+   :end-before: Information from devicetree is available to Kconfig,
+
+Information from devicetree is available to Kconfig, through the functions defined in :file:`zephyr/scripts/kconfig/kconfigfunctions.py`.
+
+The single :file:`.config` file describes the entire software configuration of the constructed binary.
+
+For detailed information about configuration options, see :ref:`zephyr:application-kconfig` in the Zephyr documentation.
+
+Memory layout configuration
+---------------------------
+
+If enabled, the :ref:`partition_manager` specific to the |NCS| provides the memory layout definition.
```

Comment:
```suggestion
If enabled, the :ref:`partition_manager`, which is specific to the |NCS|, provides the memory layout definition.
```

the x specific to y

sounds like there are multiple x, and we are just talking about the one specific to NCS.

But there is only one x.

My suggestion is not perfect either, too many commas.